File: datasets/lua-keep-incoder_6b_temp_0.2-1/HumanEval_28_concatenate.lua
-- Concatenate list of strings into a single string
-- >>> concatenate([])
-- ''
-- >>> concatenate(['a', 'b', 'c'])
-- 'abc'
local function concatenate(strings)
  return table.concat(strings, "")
end


lu = require('luaunit')

function test_humaneval()
local candidate = concatenate
    lu.assertEquals(candidate({}), '')
    lu.assertEquals(candidate({'x', 'y', 'z'}), 'xyz')
    lu.assertEquals(candidate({'x', 'y', 'z', 'w', 'k'}), 'xyzwk')
end

os.exit(lu.LuaUnit.run())

File: datasets/R-keep-code_davinci_001_temp_0.2/HumanEval_28_concatenate.R
# Concatenate list of strings into a single string
# >>> concatenate([])
# ''
# >>> concatenate(['a', 'b', 'c'])
# 'abc'
concatenate <- function(strings) {
    if (length(strings) == 0) {
        return('')
    }
    if (length(strings) == 1) {
        return(strings[1])
    }
    return(paste(strings[1], concatenate(strings[-1])))
}


test_humaneval <- function() {
candidate <- concatenate
    if(!identical(candidate(c()), '')){quit('no', 1)}
    if(!identical(candidate(c('x', 'y', 'z')), 'xyz')){quit('no', 1)}
    if(!identical(candidate(c('x', 'y', 'z', 'w', 'k')), 'xyzwk')){quit('no', 1)}
}
test_humaneval()